pyomo/contrib/doe/doe.py

```diff
@@ -32,17 +32,44 @@
 
 import pyomo.environ as pyo
 from pyomo.opt import SolverFactory
-import time
 import pickle
 from itertools import permutations, product
 import logging
-from pyomo.contrib.sensitivity_toolbox.sens import sensitivity_calculation, get_dsdp
-from pyomo.contrib.doe.scenario import Scenario_generator
+from enum import Enum
+from pyomo.common.timing import TicTocTimer
+from pyomo.contrib.sensitivity_toolbox.sens import get_dsdp
+from pyomo.contrib.doe.scenario import ScenarioGenerator, finite_difference_step
 from pyomo.contrib.doe.result import FisherResults, GridSearchResult
 
+class calculation_mode(Enum):
+    sequential_finite = 1
+    direct_kaug = 2
+
+    @classmethod
```

Comment:
@jialuw96 @jsiirola Is this typical for Enum's? @jialuw96 and I are just starting to use them in our code.